simulator/source/Combat_simulator.cpp

```diff
@@ -1384,6 +1424,20 @@ void Combat_simulator::execute_phase(Sim_state& state, bool mh_swing)
         }
     }
 
+    if (use_devastate_ && config.combat.use_devastate_in_exec_phase)
+    {
+        bool dt_ww = true;
+        if (config.combat.use_whirlwind)
+        {
+            dt_ww = std::max(time_keeper_.whirlwind_cd(), 100) > config.combat.devastate_whirlwind_cooldown_thresh;
+        }
+        if (rage >= (execute_rage_cost_ + devastate_rage_cost_) && dt_ww)
+                    {
+            devastate(state);
+            return;
+        }
+    }
```

Comment:
cp. the equivalent block in normal_phase(): the devastate block should be moved below whirlwind.

Reply:
Done